src/ooxml-viewer.ts

```diff
@@ -80,9 +82,20 @@ export class OOXMLViewer {
           await this.populateOOXMLViewer(this.zip.files, false);
 
           // set up watchers
-          const watcher: FileSystemWatcher = workspace.createFileSystemWatcher(file.fsPath);
-          watcher.onDidChange((uri: Uri) => {
-            this.reloadOoxmlFile(file.fsPath);
+          const watcher: FileSystemWatcher = workspace.createFileSystemWatcher(
+            new RelativePattern(dirname(file.fsPath), basename(file.fsPath)),
+          );
+
+          let trackingTimer: number;
```

Comment:
can we implement this more like a lock
```
let locked = false;
watcher.onDidChange(async (uri: Uri) => {
  if (!locked) {
    locked = true;
    await this.reloadOoxmlFile(file.fsPath);
    locked = false;
  }
});
```

Reply:
I think your lock solution is definitely an improvement over a timer and I tested it on some large files, and it fixes the bug. I'll have a pull request in today with this change.